### PR TITLE
fix: Handle multiple kubeconfig files like kubectl

### LINKED ISSUE #52: Handle multiple kubeconfig files
As per https://github.com/wercker/stern/issues/45#issuecomment-307221903 stern doesn't handle multiple files specified in `KUBECONFIG`.

This is [supported by kubectl](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)

Currently gives the following error:
```
(cluster-cit/kube-system) : ~ stuart.warren
$ stern kube-scheduler
failed to get client config: stat /Users/stuart.warren/.kube/config:/Users/stuart.warren/Workspace/src/github.com/stuart-warren/overview-docs/kubeconfig:/Users/stuart.warren/.kube/sso-agent-config: no such file or directory
```

### PR DESCRIPTION
**Pull request by stuart-warren**
if multiple files specified (strings split with :) load all files and
merge config

fixes #52

### CODE REVIEW

File: cmd/cli.go [hunk 1]
```diff
@@ -209,10 +210,6 @@ func getKubeConfig() (string, error) {
 		return kubeconfig, nil
 	}
 
-	if kubeconfig = os.Getenv("KUBECONFIG"); kubeconfig != "" {
```
Comment (jwolski2) [inline cmd/cli.go]: How is this change relevant to the PR?

File: kubernetes/clientset.go [hunk 2]
```diff
@@ -26,8 +28,13 @@ import (
 
 // NewClientConfig returns a new Kubernetes client config set for a context
 func NewClientConfig(configPath string, contextName string) clientcmd.ClientConfig {
+	configPaths := filepath.SplitList(configPath)
+	configLoadingRules := &clientcmd.ClientConfigLoadingRules{ExplicitPath: configPath}
+	if len(configPaths) > 1 {
```
Comment (jwolski2) [inline kubernetes/clientset.go]: Use a good ol' if-else here rather than an unnecessary reassignment. Also, what about the `len == 0` case?

Comment (stuart-warren) [reply kubernetes/clientset.go]: if len == 0 or 1 then the behavior is currently the same as it always was, using the passed in configPath variable instead which may be "" but the client library will complain as always used to.

Comment (stuart-warren) [reply kubernetes/clientset.go]: I can make that clearer and use an if-else if you really want, but personally didn't think it was clearer

File: cmd/cli.go (hunk 1 above)
Comment (stuart-warren) [reply cmd/cli.go]: this is to incrementally improve the code and reduce the amount of global state that we are accessing from functions, thereby making them more testable

Comment (jwolski2) [reply cmd/cli.go]: This change detracts from true purpose of the PR. Can you split it out into a different commit?

Comment (stuart-warren) [reply cmd/cli.go]: sorted

**Comment by jwolski2** [comment]
Thanks, @stuart-warren. I'll get some more experience stern eyes on this before merging.

File: kubernetes/clientset.go [hunk 3]
```diff
@@ -26,8 +28,15 @@ import (
 
 // NewClientConfig returns a new Kubernetes client config set for a context
 func NewClientConfig(configPath string, contextName string) clientcmd.ClientConfig {
+	configPathList := filepath.SplitList(configPath)
+	configLoadingRules := &clientcmd.ClientConfigLoadingRules{}
+	if len(configPathList) <= 1 {
```
Comment (hatchan) [inline kubernetes/clientset.go]: I was wondering why you would need this check at all. If `configPathList` == 1 then wouldn't it still work using Precedence?

**Comment by hatchan** [comment]
Thanks :+1:

**Pull request merged**